src/irr/asset/CGLTFLoader.cpp

```diff
@@ -249,66 +273,48 @@ namespace irr
 						const E_FORMAT format = getFormat(glTFAccessor.componentType.value(), glTFAccessor.type.value());
 
 						auto& glTFbufferView = glTF.bufferViews[glTFAccessor.bufferView.value()];
-						const auto& relativeOffsetInBufferView = glTFAccessor.byteOffset.value();
-						const uint32_t bufferId = glTFbufferView.buffer.value();
+						const uint32_t& bufferBindingId = glTFbufferView.buffer.value();
+						const auto& globalOffsetInBufferBindingResource = glTFbufferView.byteOffset.value();
+						const auto& relativeOffsetInBufferViewAttribute = glTFAccessor.byteOffset.value();
```

Comment:
you need to assert (not really assert, but raise an error if its not true) that this value is less than 2048

Reply:
what if attributes are packed in one binding as non interleaved? I'm writing the loader under the case where it takes place

Reply:
```json
"accessors": [
        {
            "bufferView": 0,
            "byteOffset": 0,
            "componentType": 5123,
            "count": 36,
            "max": [
                23
            ],
            "min": [
                0
            ],
            "type": "SCALAR"
        },
        {
            "bufferView": 1,
            "byteOffset": 0,
            "componentType": 5126,
            "count": 24,
            "max": [
                1.0,
                1.0,
                1.0
            ],
            "min": [
                -1.0,
                -1.0,
                -1.0
            ],
            "type": "VEC3"
        },
        {
            "bufferView": 1,
            "byteOffset": 288,
            "componentType": 5126,
            "count": 24,
            "max": [
                0.5,
                0.5,
                0.5
            ],
            "min": [
                -0.5,
                -0.5,
                -0.5
            ],
            "type": "VEC3"
        },
        {
            "bufferView": 2,
            "byteOffset": 0,
            "componentType": 5126,
            "count": 24,
            "max": [
                6.0,
                1.0
            ],
            "min": [
                0.0,
                0.0
            ],
            "type": "VEC2"
        }
    ]
```

Reply:
```json
"bufferViews": [
        {
            "buffer": 0,
            "byteOffset": 768,
            "byteLength": 72,
            "target": 34963
        },
        {
            "buffer": 0,
            "byteOffset": 0,
            "byteLength": 576,
            "byteStride": 12,
            "target": 34962
        },
        {
            "buffer": 0,
            "byteOffset": 576,
            "byteLength": 192,
            "byteStride": 8,
            "target": 34962
        }
    ],
    "buffers": [
        {
            "byteLength": 840,
            "uri": "BoxTextured0.bin"
        }
    ]
```


look at buffer at the buffer view, it is one binding

Reply:
![image](https://user-images.githubusercontent.com/34793522/96870911-dc035d00-1471-11eb-9026-44909f6693e0.png)

Reply:
no byte stride, so

Reply:
> 
> 
> what if attributes are packed in one binding as non interleaved? I'm writing the loader where various case are valid

This is possible in our engine, stride is a property of the attribute, not the buffer binding (buffer binding is only offset and buffer).

glTF's accessor's byteStride, component type and byteoffset form our ICPURenderpassIndependentPipeline's attribute and binding structs

bufferView corresponds to a `SBufferBinding<ICPUBuffer>`

Reply:
it is clear for me, but not sure if I set up the stride while doing non interleaved though - done in following commit, **check it** out when commited

Reply:
non-interleaved just means that stride equals the size of the attribute